### ISSUE TITLE
torch.median returns location even when no args passed

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#1692]
Currently `torch.median` isn't consistent in its return values with `torch.min` and `torch.max` (though all of them are retuning an aggregate and corresponding location). Bug or by design?

```python
import torch
torch.zeros(10).median()
#(
# 0
#[torch.FloatTensor of size 1]
#,
# 0
#[torch.LongTensor of size 1]
#)

torch.zeros(10).min()
#0.0
```

**Comment by lantiga** [comment]
This amounts to introducing a `medianall` function in THTensor, THCTensor, etc, mirroring what happens with `min`. @apaszke I can take this one if no one else is working on it.

**Comment by soumith** [comment]
no one's working on it, go for it!

**Comment by lantiga** [comment]
The fix should be ready for review at #1741.
There are other functions that could undergo the same treatment, namely `mode` and `kthvalue`. Do we proceed with those as well?

**Comment by vadimkantorov** [comment]
https://github.com/pytorch/pytorch/pull/1741 is merged into master, issue's solved.
